### PR TITLE
get default editor

### PR DESCRIPTION
**Pull request by asuivelentine**
#475 
Not sure what I should open? Do I get a path from FLE or do I have to use the content of it and send it to stdin, ....?
```
editor.arg(what).spawn()
````

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
You should get the contents of the `FLE` and put it into a temporary file (there's a crate for temporary files, I guess it is named `tempfile` or similar).

Make sure to also put the header in the tempfile if the user requests so.

Then call the editor with this file as argument.

---

We make it this way because I'm pretty sure that not all editors support stdin content.

**Comment by asuivelentine** [comment]
To launch a new session I need the default Terminal emulator. Any idea how to get this information? Should I use the TERM environment?

**Comment by asuivelentine** [comment]
libimagrt's function [`editor`](https://github.com/matthiasbeyer/imag/blob/master/libimagrt/src/runtime.rs#L284) returns a Command. I guess we should change this implementaiton. We need a command that starts a new terminal emulator with the editor as argument, don't we?
Maybe we could offer both implementations?

File: imag-view/src/editor.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            c.wait().unwrap_or_else(|e| {
+                error!("Error: couldn't open editor. Make sure $EDITOR is set: {}", e);
+                exit(1);
+            });
+        };
+
+        file.close().unwrap();
+    }
+}
+
```
Comment (matthiasbeyer) [inline imag-view/src/editor.rs]: I guess this whole functionality could be implemented in `libimagentryview` and therefor also implement the `Viewer` trait... so we can reuse it nicely.

Comment (asuivelentine) [inline imag-view/src/editor.rs]: Yeah, you're right #610.
I've opened a new PR to offer a loose binding between the modules.

**Comment by asuivelentine** [comment]
Ping... if your fine with that errorhandling it's meta/ready

**Comment by matthiasbeyer** [comment]
Will have a look soonish.

File: imag-view/src/main.rs [hunk 2]
```diff
@@ -75,7 +77,10 @@ fn main() {
         } else if scmd.is_present("view-in-browser") {
             warn!("Viewing in browser is currently not supported, switch to stdout");
         } else if scmd.is_present("view-in-texteditor") {
-            warn!("Viewing in texteditor is currently not supported, switch to stdout");
+            if let Err(e) = Editor::new(&entry, &rt).show() {
+                error!("Cannot view in editor: {}", e);
```
Comment (matthiasbeyer) [inline imag-view/src/main.rs]: I would like you to use `trace_error_exit()` here. Tracing the error is always better than using `error!()`, as we get the whole error chain printed.

File: imag-view/src/editor.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+impl<'a> Editor<'a> {
+    pub fn new(fle: &'a FileLockEntry, rt: &'a Runtime) -> Editor<'a> {
+        Editor{
+            fle: fle,
+            rt: rt,
+        }
+    }
+
+    pub fn show(&mut self) -> Result<(), VE>{
```
Comment (matthiasbeyer) [inline imag-view/src/editor.rs]: There's a missing whitespace before the opening bracket `{`.

Comment (matthiasbeyer) [inline imag-view/src/editor.rs]: I guess this function can also be made consuming `self` - what do you think? I have no problem if you do not want to make it consuming, though.

File: imag-view/src/editor.rs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        EditorView::new(self.rt).view_entry(self.fle.deref())
```
Comment (matthiasbeyer) [inline imag-view/src/editor.rs]: I'm not sure you need to call `deref` here - it is normally called implicitely. Try to compile it without the call to `deref` and fixup your commit if it compiles! :smile:

File: imag-view/src/editor.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+use libimagentryview::viewer::Viewer;
+use libimagentryview::error::ViewError as VE;
+
+pub struct Editor<'a> {
+    fle: &'a FileLockEntry<'a>,
+    rt: &'a Runtime<'a>,
+}
+
+impl<'a> Editor<'a> {
+    pub fn new(fle: &'a FileLockEntry, rt: &'a Runtime) -> Editor<'a> {
```
Comment (matthiasbeyer) [inline imag-view/src/editor.rs]: I would say context goes first, so please let the arguments switch places here.

**Comment by matthiasbeyer** [comment]
Due to #652, please rebase on lastest master.

File: imag-view/src/editor.rs (hunk 3 above)
Comment (asuivelentine) [inline imag-view/src/editor.rs]: Nope, your right. It's more intuitive.

**Comment by matthiasbeyer** [comment]
You consider this ready, am I right?

**Pull request merged**